pygfx/renderers/wgpu/_wgpurenderer.py

```diff
@@ -68,24 +71,105 @@ def __init__(self, *, stdinfo_uniform):
 class WgpuRenderer(Renderer):
     """A renderer that renders to a surface."""
 
-    def __init__(self, canvas):
+    def __init__(self, canvas=None, size=None, pixel_ratio=None):
+
+        # Check and normalize inputs
+        if not (canvas is None or isinstance(canvas, wgpu.gui.WgpuCanvasBase)):
+            raise TypeError("The given canvas must be None or a wgpu Canvas.")
         self._canvas = canvas
+        if not (size is None or (isinstance(size, (tuple, list)) and len(size) == 2)):
+            raise TypeError("The given size must be None or a 2-tuple.")
+        self._logical_size = (float(size[0]), float(size[1])) if size else None
+        self.pixel_ratio = pixel_ratio
 
-        self._pipelines = []
+        # Do we have enough info?
+        if canvas is None and size is None:
+            raise ValueError("WgpuRenderer needs either a canvas or a size.")
 
+        # Create adapter and device objects
         self._adapter = wgpu.request_adapter(
             canvas=canvas, power_preference="high-performance"
         )
         self._device = self._adapter.request_device(extensions=[], limits={})
 
-        self._swap_chain = self._device.configure_swap_chain(
-            canvas,
-            wgpu.TextureFormat.bgra8unorm_srgb,
+        # Prepare render targets. These are placeholders to set during
+        # each renderpass, intended to keep together the texture-view
+        # object, its size, and its format. The final (canvas) texture
+        # has bgra format, because that's what hardware seems to like,
+        # and srgb for perseptive color mapping.
+        self._canvas_texture = RenderTexture(wgpu.TextureFormat.bgra8unorm_srgb)
+        self._depth_texture = RenderTexture(wgpu.TextureFormat.depth32float)
+        # We keep track of multiple render textures, one for each post
+        # processing step, plus one final texture. Will be filled as needed.
+        self._render_textures = []
```

Comment:
For post processing, wouldn't a pair of texture suffice? You can go back and forth between them for each step.

Reply:
Great idea!

Reply:
done